typescript/src/schema/api.ts

```diff
@@ -32,19 +32,28 @@ export class API {
   hostName?: string;
   port?: string;
   mainServiceName?: string;
+  //This field is for users passing roper module name for system test.
+  publishName: string;
   // oauth_scopes: plugin.google.protobuf.IServiceOptions.prototype[".google.api.oauthScopes"];
   // TODO: subpackages
 
   constructor(
     fileDescriptors: plugin.google.protobuf.IFileDescriptorProto[],
     packageName: string,
-    grpcServiceConfig: plugin.grpc.service_config.ServiceConfig
+    grpcServiceConfig: plugin.grpc.service_config.ServiceConfig,
+    publishName?: string
   ) {
     this.naming = new Naming(
       fileDescriptors.filter(
         fd => fd.package && fd.package.startsWith(packageName)
       )
     );
+    // users specify the actual package name, if not, set it to product name.
+    if (publishName) {
+      this.publishName = publishName;
```

Comment:
```ts
this.publishName = publishName ||  this.naming.productName.toKebabCase();
```

Reply:
Thanks, this is cleaner and more elegant!